src/app/public/modules/overlay/overlay.service.ts

```diff
@@ -98,13 +106,27 @@ export class SkyOverlayService {
     }
   }
 
-  private createHostComponent(): void {
-    SkyOverlayService.host = this.dynamicComponentService.createComponent(SkyOverlayHostComponent);
-  }
+  private createOverlay(config?: SkyOverlayConfig): ComponentRef<SkyOverlayComponent> {
+    const injector = Injector.create({
+      parent: this.injector,
+      providers: [{
+        provide: SkyOverlayContext,
+        useValue: new SkyOverlayContext(config)
+      }]
+    });
+
+    const componentRef = this.componentFactoryResolver
+      .resolveComponentFactory(SkyOverlayComponent)
+      .create(injector);
+
+    this.applicationRef.attachView(componentRef.hostView);
+
+    const domElem = (componentRef.hostView as EmbeddedViewRef<any>)
+      .rootNodes[0] as HTMLElement;
+
+    document.body.appendChild(domElem);
```

Comment:
Removed the useless host component and migrated its "attach" method to the service since we need to inject providers into the overlay component.